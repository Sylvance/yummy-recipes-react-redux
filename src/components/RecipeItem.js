import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import { Button, 
  Modal, 
  ModalHeader, 
  ModalBody, 
  ModalFooter, 
  ListGroup, 
  ListGroupItem, 
  ListGroupItemHeading, 
  ListGroupItemText, 
  Badge } from 'reactstrap';
import RecipeEditFormInput from './RecipeEditFormInput'

export default class RecipeItem extends Component {
  static propTypes = {
    categoryid: PropTypes.number.isRequired,
    recipe: PropTypes.object.isRequired,
    editRecipe: PropTypes.func.isRequired,
    deleteRecipe: PropTypes.func.isRequired,
    fetchRecipes: PropTypes.func.isRequired
  }

  state = {
    editing: false,
    categoryid: this.props.categoryid,
    modal: false
  }

  toggle() {
    this.setState({
      modal: !this.state.modal
    });
  }

  handleDoubleClick = () => {
    this.setState({ editing: true })
    this.toggle()
  }

  handleSave = (id, categoryid, title, description) => {
    this.props.editRecipe(id, categoryid, title, description)
    this.props.fetchRecipes(categoryid)
    this.setState({ editing: false })
    this.toggle()
  }

  render() {
    const { recipe, deleteRecipe } = this.props

    let element
    if (this.state.editing) {
      element = (
        <Modal isOpen={this.state.modal} toggle={this.toggle} className={this.props.className}>
          <ModalHeader toggle={this.toggle}>Edit Recipe</ModalHeader>
          <ModalBody>
            <RecipeEditFormInput onSave={this.handleSave}
                        id={recipe.id}
                        categoryid={this.state.categoryid}
                        title={recipe.title}
                        description={recipe.description} />
          </ModalBody>
          <ModalFooter>
            <Button color="danger" onClick={this.toggle}>Cancel</Button>
          </ModalFooter>
        </Modal>
      )
    } else {
      element = (
        <ListGroup>
          <ListGroupItem>
            <ListGroupItemHeading onDoubleClick={this.handleDoubleClick}>
              {recipe.title}
            </ListGroupItemHeading>
            <ListGroupItemText>
              {recipe.description}
            </ListGroupItemText>
            <Badge pill color='danger' onClick={() => {
                                            deleteRecipe(recipe.id, this.state.categoryid)
                                            this.props.fetchRecipes(this.state.categoryid)
                                        }}>
              Delete
            </Badge>
          </ListGroupItem>
        </ListGroup>
      )
    }

    return (
      <li className={classnames({
        editing: this.state.editing
      })}>
        {element}
      </li>
    )
  }
}
