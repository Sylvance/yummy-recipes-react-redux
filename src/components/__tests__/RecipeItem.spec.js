// Auto-generated do not edit

/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import RecipeItem from '../RecipeItem';


describe('RecipeItem test', () => {
  it('RecipeItem should match snapshot', () => {
    const component = renderer.create(
      <RecipeItem
        key={1} categoryid={1} recipe={{}} actions={{}}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
