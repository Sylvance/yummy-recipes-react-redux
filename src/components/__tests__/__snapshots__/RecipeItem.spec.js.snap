// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeItem test RecipeItem should match snapshot 1`] = `
<li
  className=""
>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
    >
      <h5
        className="list-group-item-heading"
        onDoubleClick={[Function]}
      />
      <p
        className="list-group-item-text"
      />
      <span
        className="badge badge-danger badge-pill"
        onClick={[Function]}
      >
        Delete
      </span>
      <a
        href="/category"
      >
         Back to categories
      </a>
    </li>
  </ul>
</li>
`;
